modules/bibauthorid/lib/bibauthorid_hoover.py

```diff
@@ -0,0 +1,746 @@
+            rec[2] for rec in self.claimed_paper_signatures)
+        self.unclaimed_paper_records = set(
+            rec[2] for rec in self.unclaimed_paper_signatures)
+        self.pid = pid
+
+    def vacuum_signature(self, signature):
+        if signature not in self.unclaimed_paper_signatures and signature not in self.claimed_paper_signatures:
+            if signature[2] in self.claimed_paper_records:
+                raise DuplicateClaimedPaperException(
+                    "Vacuum a duplicated claimed paper",
+                    self.pid,
+                    signature,
+                    filter(
+                        lambda x: x[2] == signature[2],
+                        self.claimed_paper_signatures))
+
+            duplicated_signatures = filter(
+                lambda x: signature[2] == x[2],
+                self.unclaimed_paper_signatures)
+
+            if duplicated_signatures:
+                write_message(("Conflict in pid {0}"
+                              " with signature {1}".format(signature, self.pid)),
+                              verbose=4)
+                new_pid = get_free_author_id()
+                write_message(
+                    ("Moving  conflicting signature {0} from pid {1}"
+                    " to pid {2}".format(duplicated_signatures[0],
+                        self.pid,new_pid)),
+                    verbose=3)
+                HooverStats.move_signature_calls += 1
+                move_signature(duplicated_signatures[0], new_pid)
+                HooverStats.move_signature_calls += 1
+                move_signature(signature, self.pid)
+                after_vacuum = (sig[1:4]
+                                for sig in get_papers_of_author(self.pid))
+
+                if signature not in after_vacuum:
+                    HooverStats.move_signature_calls += 1
+                    move_signature(duplicated_signatures[0], self.pid)
+
+                raise DuplicateUnclaimedPaperException(
+                    "Vacuum a duplicated unclaimed paper",
+                    new_pid,
+                    signature,
+                    duplicated_signatures)
+
+            write_message("Hoovering {0} to pid {1}".format(signature, self.pid), verbose=3)
+            HooverStats.move_signature_calls += 1
+            move_signature(signature, self.pid)
+
+
+def get_signatures_with_inspireID(inspireid):
+    """get and vacuum of the signatures that belong to this inspireID
+
+    Arguments:
+    inspireID -- the string containing the inspireID
+    """
+    return get_signatures_with_inspireID_cache(inspireid)
+
+
+def get_records_with_tag(tag):
+    """return all the records with a specific tag
+
+    Arguments:
+    tag -- the tag to search for
+    """
+    assert tag in ['100__i', '100__j', '700__i', '700__j']
+    if tag.startswith("100"):
+        return run_sql(
+            "select id_bibrec from bibrec_bib10x where id_bibxxx in (select id from bib10x where tag=%s)",
```

Comment:
Be careful: use the `search_engine.perform_request_search` API. It is as fast, and does not return e.g. deleted records or records not in HEP.